### PR TITLE
Implement INFO command

### PR DESCRIPTION
**Pull request by internet-catte**
```
>>> INFO
:server1.test 371 a :Sable --
:server1.test 371 a :
:server1.test 371 a :Copyright (c) 2021-2024 Sable Development Team
:server1.test 371 a :
:server1.test 371 a :This program is free software: you can redistribute it and/or modify
:server1.test 371 a :it under the terms of the GNU Affero General Public License as published
:server1.test 371 a :by the Free Software Foundation, either version 3 of the License, or
:server1.test 371 a :(at your option) any later version.
:server1.test 371 a :
:server1.test 371 a :Source code is available at: https://github.com/Libera-Chat/sable
:server1.test 371 a :
:server1.test 371 a :A list of contributors can be found in the git log.
:server1.test 371 a :
:server1.test 371 a :Visit the Sable website at: https://sable.chat/
:server1.test 371 a :Visit us on IRC at irc.libera.chat #libera-dev
:server1.test 371 a :
:server1.test 371 a :Version: sable-0.1.0-aabc527319a4e9ff2dfb8d1e45ee35e3bbf0116f-dirty
:server1.test 371 a :Commit date: Thu, 23 May 2024 13:48:00 +0000
:server1.test 374 a :End of /INFO list
```

### CODE REVIEW

File: AUTHORS
Comment (internet-catte) [inline AUTHORS:1]: this needs actual content before merge, what/who should go here?

File: sable_ircd/src/command/handlers/info.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    let info_text = format!(
+        "Sable --
+
+Copyright (c) 2021-2024 Sable Development Team
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published
+by the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
```
Comment (internet-catte) [inline sable_ircd/src/command/handlers/info.rs:24]:
```suggestion
(at your option) any later version.

Source code is available at https://github.com/Libera-Chat/sable
```

File: sable_ircd/info.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+it under the terms of the GNU Affero General Public License as published
+by the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Source code is available at: https://github.com/Libera-Chat/sable
+
+Development is led by:
+
+spb <todo@example.com>
+val <2do@example.com>
```
Comment (internet-catte) [inline sable_ircd/info.txt:15]: who should go here?

File: sable_ircd/info.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Sable --
+
+Copyright (c) 2021-2024 Sable Development Team
```
Comment (internet-catte) [inline sable_ircd/info.txt:3]: is this the correct copyright assignment?

File: sable_ircd/info.txt (hunk 2 above)
Comment (spb) [reply sable_ircd/info.txt:15]: Personally I'd just skip this part

File: sable_ircd/info.txt (hunk 3 above)
Comment (spb) [reply sable_ircd/info.txt:3]: Good question. It's generic enough to serve the purpose for now.

**Comment by spb** [comment]
> ```
> :server1.test 371 a :Build date: Wed, 22 May 2024 22:12:16 +0000
> ```

I don't think build date is actually what we want here - the commit date would be far more useful.

**Comment by internet-catte** [comment]
should the build time be used if the tree is dirty?

**Comment by spb** [comment]
No, unless you can think of a way it would be meaningfully useful. Commit date is useful because if you're trying to quickly identify which servers do or don't have a given fix applied, you can filter on commit date without needing the git log to know which commit IDs contain the one you're after. Build date doesn't tell you that and can be actively misleading if you're unlucky.

**Pull request merged**